src/tree/ResourceGroupTreeItem.ts

```diff
@@ -16,9 +16,19 @@ import { GroupTreeItemBase } from "./GroupTreeItemBase";
 export class ResourceGroupTreeItem extends GroupTreeItemBase {
     public static contextValue: string = 'azureResourceGroup';
 
-    constructor(parent: AzExtParentTreeItem, config: GroupNodeConfiguration, public data: ResourceGroup) {
+    public data?: ResourceGroup;
+
+    constructor(parent: AzExtParentTreeItem, config: GroupNodeConfiguration, data?: ResourceGroup | Promise<ResourceGroup[]>) {
```

Comment:
Discussed offline, but should pass in a `dataTask` into constructor that is always a `Promise<ResourceGroup[]>`.  Might be worth to just make an async constructor as well so we don't run into weird issues if people are trying to mess with the ResourceGroupTreeItem before it resolves the data.  Or make an async `getData` command that always gets called before anything that relies on data being there.